python/cudf/dataframe/dataframe.py

```diff
@@ -493,6 +493,37 @@ def __deepcopy__(self, memo={}):
             memo = {}
         return self.copy()
 
+    def __add__(self, other):
+        """
+        Parameters
+        ----------
+        other, DataFrame
+            the DataFrame to be added
+
+        Returns
+        -------
+        DataFrame
+        """
+        df = DataFrame()
+
+        matched_cols = list(set(self.columns) & set(other.columns))
+        extra_self = list(set(self.columns) - set(other.columns))
+        extra_other = list(set(other.columns) - set(self.columns))
```

Comment:
I wonder if converting to a set and back affects ordering in some unpredictable way.

Reply:
Right! using `set` may affect the ordering... i will update the code.